src/charm.py

```diff
@@ -71,6 +67,20 @@ class COSConfigCharm(CharmBase):
     def __init__(self, *args):
         super().__init__(*args)
 
+        # Path to the repo in the _charm_ container, which is needed for instantiating
+        # PrometheusRulesProvider with the rule files (otherwise would need to fetch via pebble
+        # every time).
+        # Using model.storages is tricky because it only works after storage-attached event
+        # (otherwise: IndexError: list index out of range), which complicates things.
+        # So hard-coding the path to circumvent that.
+        # self._git_sync_mount_point = "/var/lib/juju/storage/content-from-git/0"
+        if len(self.model.storages["content-from-git"]) == 0:
+            # Storage isn't available yet. Since storage becomes available early enough, no need
+            # to observe storage-attached and complicate things; simply abort until it is ready.
+            return
```

Comment:
Shouldn't we set a `WaitingStatus` here?

Reply:
Iirc, at this early stage of the startup sequence juju we get maintenance status as it is.